### ISSUE TITLE
KeyError when trying to run scio-analyze on stdin

### ISSUE THREAD

**Issue by plison** [mnemonic-no/act-scio2#23]
I'm currently trying to get act-scio2 to work a number of text documents, following the README instructions.  The installation of the various packages went well, but I do get an error when running scio-analyze on some stdin input like in the provided example:

```
$ echo "The companies in the Bus; Finanical, Aviation and Automobile industry are large." | scio-analyze --beanstalk=
[2020-06-24 12:00:19] app=scio-analyze level=INFO msg=Waiting for work on stdin
[2020-06-24 12:00:19] app=scio-analyze level=ERROR msg=Got LookupError. If nltk data is missing, run scio-nltk-download, which should download all nltk data to ~/nltk_data.
[2020-06-24 12:00:19] app=scio-analyze level=ERROR msg=Task exception was never retrieved
future: <Task finished coro=<async_main() done, defined at /nr/samba/user/plison/anaconda3/lib/python3.7/site-packages/act/scio/analyze.py:98> exception=KeyError('threat_actor')>
Traceback (most recent call last):
  File "/nr/samba/user/plison/anaconda3/lib/python3.7/site-packages/act/scio/analyze.py", line 128, in async_main
    await task
  File "/nr/samba/user/plison/anaconda3/lib/python3.7/site-packages/act/scio/analyze.py", line 75, in analyze
    await asyncio.gather(*tasks)
  File "/nr/samba/user/plison/anaconda3/lib/python3.7/site-packages/act/scio/plugins/threatactor_pattern.py", line 28, in analyze
    ini['threat_actor']['alias'] = os.path.join(self.configdir, ini['threat_actor']['alias'])
  File "/nr/samba/user/plison/anaconda3/lib/python3.7/configparser.py", line 958, in __getitem__
    raise KeyError(key)
KeyError: 'threat_actor'
```

Any idea how to fix this?

**Comment by frbor** [comment]
Hi Pierre, 

it lookss like there are two errors. It is unable to find the nltk data and the configuration for the plugins. Have you ran these commands as stated in the readme?

```
scio-nltk-download
scio-config user
```
best regards,

  Fredrik

**Comment by plison** [comment]
Yes, I ran both commands, without errors.

**Comment by frbor** [comment]
What is the output of these commands?

```
ls -latrh ~/nltk_data
ls -latrh ~/.config/scio/
```

**Comment by plison** [comment]
It seems like everything is in place:

```
plison@samba8:~/code/oa/weak_supervision$ ls -latrh ~/nltk_data
total 24K
drwxrwxr-x  3 plison u_samba   48 Aug  8  2019 tokenizers
drwx--x--x 82 plison u_samba 8.0K Jun 19 11:04 ..
drwxrwxr-x  3 plison u_samba   72 Jun 24 11:25 chunkers
drwxrwxr-x  7 plison u_samba  296 Jun 24 11:25 grammars
drwxrwxr-x  3 plison u_samba   52 Jun 24 11:25 help
drwxrwxr-x  6 plison u_samba  316 Jun 24 11:25 taggers
drwxrwxr-x 12 plison u_samba  201 Jun 24 11:25 .
drwxrwxr-x  2 plison u_samba   39 Jun 24 11:25 sentiment
drwxrwxr-x  4 plison u_samba  125 Jun 24 11:25 stemmers
drwxrwxr-x  6 plison u_samba  228 Jun 24 11:25 models
drwxrwxr-x  4 plison u_samba  106 Jun 24 11:25 misc
drwxrwxr-x 74 plison u_samba 8.0K Jun 24 12:10 corpora
plison@samba8:~/code/oa/weak_supervision$ ls -latrh ~/.config/scio/
total 4.0K
drwx------ 36 plison u_samba 4.0K Jun 24 11:31 ..
drwxrwxr-x  4 plison u_samba   43 Jun 24 11:31 .
drwxrwxr-x  3 plison u_samba   70 Jun 24 11:31 etc
drwxrwxr-x  2 plison u_samba  119 Jun 24 11:31 vendor
```

**Comment by frbor** [comment]
Not sure what this error can be. Can you also get the output of this?

`cat ~/.config/scio/etc/plugins/threatactor_pattern.ini`

**Comment by plison** [comment]
Aha! `No such file or directory`. So that's where the error was. 

Here's the full content of the `scio/etc` directory:

```
/nr/samba/user/plison/.config/scio/:
total 4.0K
drwx------ 36 plison u_samba 4.0K Jun 24 11:31 ..
drwxrwxr-x  4 plison u_samba   43 Jun 24 11:31 .
drwxrwxr-x  3 plison u_samba   70 Jun 24 11:31 etc
drwxrwxr-x  2 plison u_samba  119 Jun 24 11:31 vendor

/nr/samba/user/plison/.config/scio/etc:
total 8.0K
drwxrwxr-x 2 plison u_samba   10 Jun 24 11:31 plugins
drwxrwxr-x 4 plison u_samba   43 Jun 24 11:31 ..
-rw-rw-r-- 1 plison u_samba  390 Jun 24 11:31 scio.ini
drwxrwxr-x 3 plison u_samba   70 Jun 24 11:31 .
-rw-rw-r-- 1 plison u_samba 1.7K Jun 24 11:31 feeds.txt

/nr/samba/user/plison/.config/scio/etc/plugins:
total 0
drwxrwxr-x 2 plison u_samba 10 Jun 24 11:31 .
drwxrwxr-x 3 plison u_samba 70 Jun 24 11:31 ..

/nr/samba/user/plison/.config/scio/vendor:
total 5.8M
drwxrwxr-x 4 plison u_samba   43 Jun 24 11:31 ..
-rw-rw-r-- 1 plison u_samba  64K Jun 24 11:31 ISO-3166-countries-with-regional-codes.json
-rw-rw-r-- 1 plison u_samba 5.7M Jun 24 11:31 cities15000.txt
drwxrwxr-x 2 plison u_samba  119 Jun 24 11:31 .
-rw-rw-r-- 1 plison u_samba  28K Jun 24 11:31 countryInfo.txt
```

**Comment by frbor** [comment]
Can you try to regenreate the config, then and copy the output here?

Like this:

```
$ rm -rf ~/.config/scio/
$ scio-config user
Config dir: /home/fredrikb/.config/scio
creating directory /home/fredrikb/.config/scio/etc
creating directory /home/fredrikb/.config/scio/etc/plugins
creating directory /home/fredrikb/.config/scio/vendor
Config copied to /home/fredrikb/.config/scio/etc/scio.ini
Config copied to /home/fredrikb/.config/scio/etc/feeds.txt
Config copied to /home/fredrikb/.config/scio/etc/plugins/sectors.cfg
Config copied to /home/fredrikb/.config/scio/etc/plugins/locations.ini
Config copied to /home/fredrikb/.config/scio/etc/plugins/tools.cfg
Config copied to /home/fredrikb/.config/scio/etc/plugins/ta_aliases.cfg
Config copied to /home/fredrikb/.config/scio/etc/plugins/threatactor_pattern.ini
Config copied to /home/fredrikb/.config/scio/etc/plugins/sectors.ini
Config copied to /home/fredrikb/.config/scio/etc/plugins/tools_pattern.ini
Config copied to /home/fredrikb/.config/scio/etc/plugins/country_aliases.cfg
Config copied to /home/fredrikb/.config/scio/vendor/cities15000.txt
Config copied to /home/fredrikb/.config/scio/vendor/ISO-3166-countries-with-regional-codes.json
Config copied to /home/fredrikb/.config/scio/vendor/countryInfo.txt

```

**Comment by plison** [comment]
Sure:
```
plison@samba8:~/code/oa/weak_supervision$ rm -rf ~/.config/scio/
plison@samba8:~/code/oa/weak_supervision$ scio-config user
Config dir: /nr/samba/user/plison/.config/scio
creating directory /nr/samba/user/plison/.config/scio/etc
creating directory /nr/samba/user/plison/.config/scio/etc/plugins
creating directory /nr/samba/user/plison/.config/scio/vendor
Config copied to /nr/samba/user/plison/.config/scio/etc/scio.ini
Config copied to /nr/samba/user/plison/.config/scio/etc/feeds.txt
Config copied to /nr/samba/user/plison/.config/scio/vendor/ISO-3166-countries-with-regional-codes.json
Config copied to /nr/samba/user/plison/.config/scio/vendor/cities15000.txt
Config copied to /nr/samba/user/plison/.config/scio/vendor/countryInfo.txt
```

**Comment by frbor** [comment]
Hmm, ok so the plugin config is never created.

We will need to look into that, but I will send you copy on my local configuration offline, so you can test it out in the mean while.

**Comment by plison** [comment]
Ok, no problem! I've tested the system with your local configuration files, and it works now.

**Comment by frbor** [comment]
We finally found a way to reproduce this issue, and it should be fixed in version 0.9 of act-scio which is published on pypi now.
